src/software/proto/repeated_any_msg.proto

```diff
@@ -0,0 +1,8 @@
+syntax = 'proto3';
+import "google/protobuf/any.proto";
+
+message RepeatedAnyMsg
+{
+    string message_type                   = 1;
+    repeated google.protobuf.Any messages = 2;
```

Comment:
have you considered: https://developers.google.com/protocol-buffers/docs/techniques#self-description

Reply:
This might be worthwhile if we're going to build a Python environment, will look into it

Reply:
Out of scope for now, will consider it for the python environment proposal. I don't think working with descriptor protos is well-supported for Python though, the linked page says `DynamicMessage` is only for C++ and Java.